packages/datadog_tracking_http_client/lib/src/tracking_http_client.dart

```diff
@@ -164,6 +164,16 @@ class DatadogTrackingHttpClient implements HttpClient {
     return request;
   }
 
+  @override
+  set connectionFactory(
+          Future<ConnectionTask<Socket>> Function(
+                  Uri url, String? proxyHost, int? proxyPort)?
+              f) =>
+      innerClient.connectionFactory = f;
+
+  @override
+  set keyLog(Function(String line)? callback) => innerClient.keyLog = callback;
```

Comment:
Just FYI, you can do something like the following code, in order to support 2.16 and 2.17
```suggestion
  // ignore: override_on_non_overriding_member
  @override
  set connectionFactory(
      Future<ConnectionTask<Socket>> Function(
              Uri url, String? proxyHost, int? proxyPort)?
          f) {
    try {
      (innerClient as dynamic).connectionFactory = f;
    } on NoSuchMethodError catch (_) { }
  }

  // ignore: override_on_non_overriding_member
  @override
  set keyLog(Function(String line)? callback) {
    try {
      (innerClient as dynamic).keyLog = callback;
    } on NoSuchMethodError catch (_) { }
  }
```

Reply:
Hrm, good call. I always get a little bit nervous about casting to dynamic, but might be a good stop-gap to make maintaining the two versions easier.

Reply:
Yeah, I agree that it's not the best solution, but it's probably worth it. Also, you can easily test this, and it also just needs to be tested on Dart >= 2.17 as the methods aren't available on earlier versions.

Reply:
You're welcome